### PR TITLE
update ESLint to v9 and other smaller dependencies

### PR DESCRIPTION
**Pull request by matyasf**
Closes: INSTUI-4099, INSTUI-4062, INSTUI-4183
to test: check docs app, check output of `npm run lint` for erroneous lint warnings. 

The good:
- `ui-eslint-config` package is gone :)
- Lots of small changes that were detected by the correctly configured linter, like `packages/template-` packages had some leftover files that could be deleted
- no weird ESLint errors when nothing was wrong
- no need for 100+ `.eslintignore` files
- it detected some duplicate tests, these were removed
- added a plugin for specifically linting `vitest` tests

The bad:
- `eslint-plugin-jsx-a11y` is not yet converted to ESLint v9, add it back when its ready https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/978
- `eslint-import-resolver-typescript` and `eslint-plugin-import-x` are not converted either, but these are less useful plugins
- Our own ESLint plugin is not converted yet to ESLint v9, now its turned off, should be converted in the near future (and wont need its own package).
- Docs app linting is not 100% yet because it has some messy interwoven browser+node code.

Other smaller changes:
- Fixed a warning in Table about the `isStacked` and `hover` prop that was always showing in the console
- Fixed a test that broke with the `moment` upgrade, turns out the test was bad and moment got better

### CODE REVIEW

File: eslint.config.mjs [hunk 1]
```diff
@@ -0,0 +1,216 @@
+/*
```
Comment (matyasf) [inline eslint.config.mjs:1]: This file was the bulk of the work

File: packages/ui-time-select/src/TimeSelect/__tests__/TimeSelect.test.tsx [hunk 2]
```diff
@@ -459,7 +459,7 @@ describe('<TimeSelect />', async () => {
     const select = await TimeSelectLocator.find()
     const input = await select.findInput()
     await input.change({ target: { value: '' } })
-    await input.typeIn('04:45:55 AM')
+    await input.typeIn('4:45:55 AM')
```
Comment (matyasf) [inline packages/ui-time-select/src/TimeSelect/__tests__/TimeSelect.test.tsx:462]: new moment is more exact in strict parsing mode what we're using

File: packages/ui-utils/src/__new-tests__/hash.test.tsx [hunk 3]
```diff
@@ -347,24 +347,6 @@ describe('hash', () => {
       expect(result1).toEqual(result2)
     })
 
-    it('hashes two different simple objects to different values', () => {
```
Comment (matyasf) [inline packages/ui-utils/src/__new-tests__/hash.test.tsx:350]: duplicate test

File: packages/ui-buttons/src/Button/__new-tests__/Button.test.tsx [hunk 4]
```diff
@@ -74,7 +74,7 @@ describe('<Button/>', () => {
     expect(button).toHaveTextContent(children)
   })
 
-  it('should render a button', () => {
+  it('should render a button with name prop', () => {
```
Comment (joyenjoyer) [inline packages/ui-buttons/src/Button/__new-tests__/Button.test.tsx:77]: `Button` does not have a `name` property. The text content `Hello World` is its child.

File: packages/ui-source-code-editor/src/SourceCodeEditor/index.tsx [hunk 5]
```diff
@@ -483,7 +483,7 @@ class SourceCodeEditor extends Component<SourceCodeEditorProps> {
     const { styles } = this.props
 
     if (!styles?.theme || !styles.highlightStyle) {
-      return
+      return undefined
```
Comment (joyenjoyer) [inline packages/ui-source-code-editor/src/SourceCodeEditor/index.tsx:486]: Why is this needed here?

**Review by HerrTopi — APPROVED** [review]
Just a thought: shouldn't we turn "Unexpected any. Specify a different type" warnings off? Those are polluting the log and serve no real info.

Otherwise: great work!

File: packages/ui-source-code-editor/src/SourceCodeEditor/index.tsx (hunk 5 above)
Comment (matyasf) [reply packages/ui-source-code-editor/src/SourceCodeEditor/index.tsx:486]: ESLint expects that a getter always returns a value. IMO this is a good rule, and its OK if we make such edge cases more explicit

File: packages/ui-buttons/src/Button/__new-tests__/Button.test.tsx (hunk 4 above)
Comment (matyasf) [reply packages/ui-buttons/src/Button/__new-tests__/Button.test.tsx:77]: good catch, I'll delete it

**Comment by matyasf** [comment]
> Just a thought: shouldn't we turn "Unexpected any. Specify a different type" warnings off? Those are polluting the log and serve no real info.

Yep, it gets a bit spammy because we have a lot of these. But I think they are useful, we should try to get rid of `any`s
